ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/generator/IdGeneratorBase.java

```diff
@@ -0,0 +1,195 @@
+package io.appform.ranger.discovery.bundle.id.generator;
+
+import com.google.common.base.Preconditions;
+import dev.failsafe.Failsafe;
+import dev.failsafe.FailsafeExecutor;
+import dev.failsafe.RetryPolicy;
+import io.appform.ranger.discovery.bundle.id.Domain;
+import io.appform.ranger.discovery.bundle.id.GenerationResult;
+import io.appform.ranger.discovery.bundle.id.Id;
+import io.appform.ranger.discovery.bundle.id.IdInfo;
+import io.appform.ranger.discovery.bundle.id.IdValidationState;
+import io.appform.ranger.discovery.bundle.id.constraints.IdValidationConstraint;
+import io.appform.ranger.discovery.bundle.id.formatter.IdFormatter;
+import io.appform.ranger.discovery.bundle.id.formatter.IdFormatters;
+import io.appform.ranger.discovery.bundle.id.nonce.NonceGeneratorBase;
+import io.appform.ranger.discovery.bundle.id.request.IdGenerationInput;
+import io.appform.ranger.discovery.bundle.id.request.IdGenerationRequest;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.joda.time.DateTime;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Base Id Generator
+ */
+@SuppressWarnings("unused")
+@Slf4j
+public class IdGeneratorBase {
+
+    @Getter
+    private static int NODE_ID;
+    private final List<IdValidationConstraint> globalConstraints = new ArrayList<>();
+    private final Map<String, Domain> registeredDomains = new ConcurrentHashMap<>(Map.of(Domain.DEFAULT_DOMAIN_NAME, Domain.DEFAULT));
+    private final FailsafeExecutor<GenerationResult> retryer;
+
+    protected final IdFormatter idFormatter;
+    protected final NonceGeneratorBase nonceGenerator;
+
+    public static void initialize(final int node) {
```

Comment:
We can get rid of this and let the library initialise per instance of this class